Simplify mixins

magicus opened the pull request:
Simplify mixins so they just send an event via EventFactory, or calls a specialized callback function.

I have not yet touched BossHealthOverlayMixin. Not sure if I dare to in this PR. :-)

magicus committed:
Simplify mixins so they just send an event via EventFactory, or calls a specialized callback function

magicus committed:
I thought I ran spotless...

P0keDev commented on common/src/main/java/com/wynntils/mc/EventFactory.java:
```diff
@@ -194,6 +211,12 @@ public static void onKeyInput(int key, int scanCode, int action, int modifiers)
         post(new KeyInputEvent(key, scanCode, action, modifiers));
     }
 
+    public static boolean onChatSend(String message) {
```
this should support editing the sent message as well; it's important for features such as chat items

maybe this could be accomplished by simply returning the event object?

Incompleteusern replied:
yeah imo `EventFactory` should either return the event or a boolean

magicus replied:
I think we need further changes in the mixin to be able to edit the message. I'm not against it, but suggest we wait with implementing it until it's needed (so we can notice if it does not work).

magicus replied:
@P0keDev Are you ok with this?

P0keDev commented on common/src/main/java/com/wynntils/mc/event/ChatSendMessageEvent.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public boolean isCancelable() {
```
you can use the `@Cancelable` annotation instead of including this override

magicus replied:
Didn't know that, thanks, will fix.

magicus committed:
Use @@Cancelable instead

magicus committed:
Merge branch 'main' into simplify-mixings

magicus committed:
Promote keybinds and command framework to core (#67)

* Promote keybinds and command framework to core

* Spotless
# Conflicts:
#	common/src/main/java/com/wynntils/mc/mixin/CommandSuggestionsMixin.java
#	common/src/main/java/com/wynntils/mc/mixin/LocalPlayerMixin.java

magicus committed:
Merge branch 'main' into simplify-mixings

kristofbolyai reviewed (changes requested):
Fix the merge, the PR which moved keybinds to core was not merged correctly, an import still refers to the old package, otherwise LGTM

magicus committed:
Fix merge

magicus committed:
Fix merge, part 2

kristofbolyai commented:
Also run spotless.
